collectors/htcondor/collector.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+
+import logging
+
+from datetime import datetime as dt
+from typing import Optional, Tuple, List
+from urllib.parse import quote
+from asyncio import create_subprocess_shell
+from asyncio.subprocess import PIPE
+
+from pyauditor import (
+    AuditorClientBuilder,
+    AuditorClient,
+    Record,
+    Component,
+    Score,
+    Meta,
+)
+
+
+from utils import maybe_convert, get_value
+from config import Config
+from state_db import StateDB
+from exceptions import RecordGenerationException
+
+
+class CondorHistoryCollector(object):
+    def __init__(self, config: Config):
+        self.config = config
+        self.logger = self.setup_logger()
+        self.client = self.setup_auditor_client()
+        self.state_db = StateDB(config.state_db)
+
+    def setup_auditor_client(self) -> AuditorClient:
+        """Sets up the auditor client."""
+        builder = AuditorClientBuilder().address(self.config.addr, self.config.port)
```

Comment:
Could it be useful to add a `timeout`?

Reply:
This is addressed in [573b520](https://github.com/ALU-Schumacher/AUDITOR/pull/264/commits/573b5208826228ac336cab8049d49adb7d306f86).